opentelemetry-otlp/src/metric.rs

```diff
@@ -160,38 +154,60 @@ where
         }
     }
 
+    /// Build with the given temporality selector
+    pub fn with_temporality_selector<T: TemporalitySelector + 'static>(self, selector: T) -> Self {
+        OtlpMetricPipeline {
+            temporality_selector: Some(Box::new(selector)),
+            ..self
+        }
+    }
+
+    /// Build with the given aggregation selector
+    pub fn with_aggregation_selector<T: AggregationSelector + 'static>(self, selector: T) -> Self {
+        OtlpMetricPipeline {
+            aggregator_selector: Some(Box::new(selector)),
+            ..self
+        }
+    }
+
     /// Build push controller.
-    pub fn build(self) -> Result<BasicController> {
+    pub fn build(self) -> Result<MeterProvider> {
         let exporter = self
             .exporter_pipeline
             .ok_or(Error::NoExporterBuilder)?
-            .build_metrics_exporter(Box::new(self.temporality_selector.clone()))?;
+            .build_metrics_exporter(
+                self.temporality_selector
+                    .unwrap_or_else(|| Box::new(DefaultTemporalitySelector::new())),
+                self.aggregator_selector
+                    .unwrap_or_else(|| Box::new(DefaultAggregationSelector::new())),
+            )?;
+
+        let mut builder = PeriodicReader::builder(exporter, self.rt);
 
-        let mut builder = controllers::basic(processors::factory(
-            self.aggregator_selector,
-            self.temporality_selector,
-        ))
-        .with_exporter(exporter);
         if let Some(period) = self.period {
-            builder = builder.with_collect_period(period);
+            builder = builder.with_interval(period);
         }
         if let Some(timeout) = self.timeout {
-            builder = builder.with_collect_timeout(timeout)
+            builder = builder.with_timeout(timeout)
         }
+
+        let reader = builder.build();
```

Comment:
If we are constructing our own `MeterProvider` is it possible to build the stand alone otlp `Reader`? For instance if we wanted both Prometheus and Otlp metrics export.

Reply:
Yeah all exporters should provide a way to build just their exporter, and exporters should implement either `MetricReader` or `PushMetricExporter`. If they don't currently we should fix that.